controllers/teleport_controller/teleport_controller.cpp

```diff
@@ -0,0 +1,147 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2021 NUbots <nubots@nubots.net>
+ */
+
+#include <RobotisOp2MotionManager.hpp>
+#include <fstream>
+#include <iostream>
+#include <random>
+#include <webots/Supervisor.hpp>
+
+#include "yaml-cpp/yaml.h"
+
+// AxisAngle rotations will be read from a config file and saved here
+std::vector<std::vector<double>> rotations;
+// Preset motions for the robot will be read and saved here
+std::vector<int> playPages;
+
+int main() {
+
+    // Load config file
+    try {
+        YAML::Node config = YAML::LoadFile("config.yaml");
+        rotations         = config["rotations"].as<std::vector<std::vector<double>>>();
+        playPages         = config["playpages"].as<std::vector<int>>();
+    }
+    catch (const std::string& e) {
+        std::cout << e << std::endl;
+        return 0;
+    }
+
+    // create the Supervisor instance and assign it to a robot
+    webots::Supervisor supervisor = webots::Supervisor();
+    webots::Node& target          = *supervisor.getFromDef("RED_1");
+
+    // Get the time step of the current world.
+    int timeStep = int(supervisor.getBasicTimeStep());
+
+    // Add the motion manager to make the robot stand up and look around
+    managers::RobotisOp2MotionManager mMotionManager = managers::RobotisOp2MotionManager(&supervisor);
+    mMotionManager.playPage(1);  // Look around
+
+    // Generate random seed
+    std::random_device rd;   // Will be used to obtain a seed for the random number engine
+    std::mt19937 gen(rd());  // Standard mersenne_twister_engine seeded with rd()
+    std::uniform_int_distribution<> xDistrib(0, 1080);
+    std::uniform_int_distribution<> yDistrib(0, 1600);
+
+    while (supervisor.step(timeStep) != -1) {
+
+        // Grab the current translation field of the robot to modify
+        webots::Field& target_translation_field = *target.getField("translation");
+        // Convert the field to a vector to output to console
+        const double* target_translation_vec = target_translation_field.getSFVec3f();
+
+        // Output current location
+        try {
+            std::ofstream log;
+            log.open("log.txt", std::fstream::app);
+            log << "Location: " << std::endl;
+            log << "X: " << target_translation_vec[0];
+            log << " Y: " << target_translation_vec[1];
+            log << " Z: " << target_translation_vec[2] << std::endl;
+            log.close();
+        }
+        catch (const std::string& e) {
```

Comment:
```suggestion
        catch (const YAML::BadFile& e) {
```

Reply:
This isn't a YAML file tho, just a txt. Is it still the same exception?

Reply:
This isn't a YAML file tho, just a txt. Is it still the same exception?